packages/app-builder/src/routes/ressources+/settings+/data-model+/tables+/$tableId+/exported-fields+/delete.ts

```diff
@@ -0,0 +1,93 @@
+import { ExportedFields } from '@app-builder/models/data-model';
+import { ingestedDataFieldSchema } from '@app-builder/queries/settings/scenarios/schema';
+import invariant from 'tiny-invariant';
+import z from 'zod';
+
+const exportedFieldSchema = z.union([
+  z.object({ triggerObjectField: z.string() }),
+  z.object({ ingestedDataField: ingestedDataFieldSchema }),
+]);
+
+type GetExportedFieldsRessourceActionResult = ServerFnResult<
+  | Response
+  | {
+      success: boolean;
+      errors?: string[];
+      data?: ExportedFields;
+    }
+>;
+
+export const action = createServerFn(
+  [handleRedirectMiddleware, authMiddleware],
+  async function getExportedFieldsRessourceAction({
+    request,
+    context,
+    params,
+  }): GetExportedFieldsRessourceActionResult {
+    if ('DELETE' !== request.method) {
+      return { success: false, errors: ['Method not allowed'] };
+    }
+    const { toastSessionService } = context.services;
+    const toastSession = await toastSessionService.getSession(request);
+
+    const { tableId } = params;
+    invariant(tableId, 'Table ID is required');
+
+    const body = await request.json();
+
+    const parsedCreate = exportedFieldSchema.safeParse(body);
+    if (!parsedCreate.success) {
+      return { success: false, errors: ['Invalid payload'] };
+    }
+
+    try {
+      const current =
+        await context.authInfo.dataModelRepository.getDataModelTableExportedFields(tableId);
+
+      if ('triggerObjectField' in parsedCreate.data) {
+        const field = parsedCreate.data.triggerObjectField;
+        if (current.triggerObjectFields.includes(field)) {
+          return { success: false, errors: ['Field already exported'] };
+        }
+        const exportedFields =
+          await context.authInfo.dataModelRepository.updateDataModelTableExportedFields(tableId, {
+            triggerObjectFields: [...current.triggerObjectFields, field],
+            ingestedDataFields: current.ingestedDataFields,
+          });
+        return { success: true, data: exportedFields };
+      }
+
+      if ('ingestedDataField' in parsedCreate.data) {
+        const field = parsedCreate.data.ingestedDataField;
+        const exists = (current.ingestedDataFields ?? []).some(
+          (f) => f.name === field.name && (f.path ?? []).join('.') === field.path.join('.'),
+        );
+        if (exists) {
+          return { success: false, errors: ['Field already exported'] };
+        }
+        const exportedFields =
+          await context.authInfo.dataModelRepository.updateDataModelTableExportedFields(tableId, {
+            triggerObjectFields: current.triggerObjectFields,
+            ingestedDataFields: [...(current.ingestedDataFields ?? undefined), field],
+          });
+        return { success: true, data: exportedFields };
+      }
+
+      return { success: false, errors: ['Invalid payload'] };
+    } catch (err) {
+      console.error(err);
```

Comment:
Is the console.log still wanted?

Reply:
I thought this error should never happen, but it would be nice to log it just in case.